geopandas/tests/test_geodataframe.py

```diff
@@ -457,6 +457,13 @@ def test_to_json_drop_id_only_geom_column(self):
         for f in data["features"]:
             assert "id" not in f.keys()
 
+    def test_to_json_with_duplicate_columns(self):
```

Comment:
We should add a test to `iterfeatures` section (around line 644) since the change happens there. Then we don't necessarily need them in `to_json` and `to_file`.

Reply:
Thanks for your feedback. I think is best if we keep the test in `to_json` and `to_file` just for completeness